### PR TITLE
feat: unify metadata on gallery detail page

### LINKED ISSUE #8805: Video NFTs not loading at all on iOS
Videos not loading on the phone at all. On my phone, I opened [this](https://kodadot.xyz/ahk/gallery/126-20) video NFT -  is not working.
- Production, AHK
May be related to #8729 - but the issue is a bit different because the video NFT can be loaded on a desktop, but not on a phone. 
IMG_9943

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Related https://github.com/kodadot/workers/pull/206
- [x] Fix #8805
- [x] Fix https://github.com/kodadot/workers/issues/187
- [x] Fix https://github.com/kodadot/workers/issues/185

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)

## Screenshot 📸

- [x] My fix has changed UI

tested on: https://deploy-preview-8171--koda-canary.netlify.app/ahp/gallery/10-26

image

### CODE REVIEW

**Comment by vikiival** [comment]
@preschian any update with this one?

**Comment by preschian** [comment]
> @preschian any update with this one?

on it, will add test case in here https://github.com/kodadot/workers/pull/206 first

**Review by Jarsen136 — APPROVED** [review]

File: components/gallery/GalleryItem.vue [hunk 1]
```diff
@@ -216,8 +216,15 @@ const preferencesStore = usePreferencesStore()
...
-  galleryItem
+const {
+  nft,
+  nftMetadata,
+  nftImage,
+  nftAnimation,
+  nftAnimationMimeType,
+  nftMimeType,
+  nftResources,
+} = galleryItem
```
Comment (vikiival) [inline components/gallery/GalleryItem.vue:227]: why this `nft` prefix?

**Comment by prury** [comment]
- i think image loading on gallery item takes a bit more time than on canary: 

https://github.com/kodadot/nft-gallery/assets/36627808/9193ab2b-deb5-41a5-b567-5637be9d149f

- saw that the `#number` after item names is now gone ✔️ 

- if i am redirecting user to a `.AVIF`, why does it say its a PNG?
image

- Kodachain does not load here for me(no buffering):
https://deploy-preview-8171--koda-canary.netlify.app/ahp/gallery/10-26

**Comment by preschian** [comment]
> * i think image loading on gallery item takes a bit more time than on canary:

noted

> * saw that the `#number` after item names is now gone ✔️

meanwhile, we have this https://github.com/kodadot/nft-gallery/issues/8300. not sure which one is correct. I think it's better to adjust the serial number from the indexer side instead

> * if i am redirecting user to a `.AVIF`, why does it say its a PNG?

`AVIF` is our optimized content with Cloudflare. original metadata is `PNG` for that item. https://github.com/kodadot/nft-gallery/issues/4551

> * Kodachain does not load here for me(no buffering, but the Animated Media link inside description tab works fine):
>   [deploy-preview-8171--koda-canary.netlify.app/ahp/gallery/10-26](https://deploy-preview-8171--koda-canary.netlify.app/ahp/gallery/10-26)

Interesting, this one should not be affected by this PR. At the moment, I will put this PR in the draft instead. Will revisit this PR once this one is implemented https://github.com/kodadot/workers/issues/187

**Comment by preschian** [comment]
> * i think image loading on gallery item takes a bit more time than on canary:
> * saw that the `#number` after item names is now gone ✔️

updated

**Review by roiLeo — APPROVED** [review]

**Review by roiLeo — COMMENTED** [review]
doesn't fix Ios but there is some progress

Capture d’écran 2024-01-05 à 9 39 14 AM

**Comment by vikiival** [comment]
Maybe one last round from prury?

**Comment by yangwao** [comment]
cc @prury

**Comment by prury** [comment]
https://deploy-preview-8171--koda-canary.netlify.app/ahk/gallery/315-9 (firefox, only loads after a few refreshes)
https://deploy-preview-8171--koda-canary.netlify.app/ahk/gallery/126-18 (firefox only)

image


https://github.com/kodadot/nft-gallery/assets/36627808/5b5ac5f6-96f9-44e8-9595-4e7bcc27df7c


on chrome, it takes a bit of time but it loads on explorer, on firefox it never loads and shows no console errors

image

verdict: 
i think PR solves most of problems, only remaining ones are from firefox

**Comment by yangwao** [comment]
> only remaining ones are from firefox desktop

follow up issue and we will have look

**Comment by yangwao** [comment]
Thanks!
pay 80 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $80 @ 7.45 USD/DOT ~ 10.738 $DOT
🧗 [1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p](https://polkadot.subscan.io/account/1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)
🔗 [0xb25f8fc4dcbc50e7add40d307bc5cd28d824aea4eacc3e8b60f09354769cdaf3](https://polkadot.subscan.io/extrinsic/0xb25f8fc4dcbc50e7add40d307bc5cd28d824aea4eacc3e8b60f09354769cdaf3)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**